packages/@aws-cdk/aws-sagemaker/lib/model.ts

```diff
@@ -0,0 +1,334 @@
+    private readonly primaryContainer: IContainerDefinition;
+    private readonly containers: IContainerDefinition[] = [];
+
+    constructor(scope: Construct, id: string, props: ModelProps = {}) {
+        super(scope, id);
+
+        // set the model if if defined
+        if (props.modelName) { this.modelName = props.modelName; }
+        if (props.primaryContainer) { this.primaryContainer = props.primaryContainer; }
+
+        // configure networking
+        if (props.vpc) {
+            this.vpc = props.vpc;
+            // create a security group
+            this.securityGroup = new ec2.SecurityGroup(this, 'ModelSecurityGroup', {
+                vpc: props.vpc
+            });
+            this.connections = new ec2.Connections({ securityGroups: [this.securityGroup] });
+            this.securityGroups.push(this.securityGroup);
+            this.subnets = props.vpc.selectSubnets(props.vpcSubnets);
+        }
+
+        // set the sagemaker role or create new one
+        this.role = props.role || new iam.Role(this, 'SagemakerRole', {
+            assumedBy: new iam.ServicePrincipal('sagemaker.amazonaws.com'),
+            managedPolicies: [iam.ManagedPolicy.fromAwsManagedPolicyName('service-role/AmazonSageMakerFullAccess')],
+        });
+
+        this.node.applyAspect(new Tag(NAME_TAG, this.node.path));
+
+        const model = new CfnModel(this, 'Model', {
+            executionRoleArn: this.role.roleArn,
+            modelName: this.modelName,
+            primaryContainer: Lazy.anyValue({ produce: () => (this.containers.length === 0) ?
+                this.renderPrimaryContainer(scope, this.primaryContainer) : undefined }),
+            vpcConfig: Lazy.anyValue({ produce: () => this.renderVpcConfig() }),
+            containers: Lazy.anyValue({ produce: () => this.renderContainerList(scope, this.containers) })
+        });
+        this.modelName = model.attrModelName;
+        this.modelArn = Stack.of(this).formatArn({
+            service: 'sagemaker',
+            resource: 'model',
+            resourceName: this.modelName
+          });
+    }
+
+    /**
+     * Add the security group to all instances via the launch configuration
+     * security groups array.
+     *
+     * @param securityGroup: The security group to add
+     */
+    public addSecurityGroup(securityGroup: ec2.ISecurityGroup): void {
+        this.securityGroups.push(securityGroup);
+    }
+
+    /**
+     * Add the container definition to the model.
+     *
+     * @param container: The container to add
+     */
+    public addContainer(container: IContainerDefinition): void {
+        this.containers.push(container);
+    }
+
+    protected validate(): string[] {
+        const result = super.validate();
+        // check that either primary container or list of containers defined
+        if (!(this.primaryContainer) && (this.containers.length === 0)) {
+            result.push("Must define either Primary Container or list of inference containers");
```

Comment:
This message is a little ambiguous - perhaps "A SageMaker model must define either a primary container, or a list of inference containers".